x-pack/plugins/cases/common/types/api/configure/v1.ts

```diff
@@ -21,7 +21,11 @@ export const CustomFieldConfigurationWithoutTypeRt = rt.strict({
   /**
    * key of custom field
    */
-  key: limitedStringSchema({ fieldName: 'key', min: 1, max: MAX_CUSTOM_FIELD_KEY_LENGTH }),
+  key: regexStringRt({
+    codec: limitedStringSchema({ fieldName: 'key', min: 1, max: MAX_CUSTOM_FIELD_KEY_LENGTH }),
+    pattern: '^[a-z0-9_-]+$',
```

Comment:
We could also have `^[a-z0-9_-]{1,MAX_CUSTOM_FIELD_KEY_LENGTH}$`.

We don't do it so we can have more specific error messages?

Reply:
I was not aware of that! Let me try that.

Reply:
Probably have to escape the `$` at the end though